plugins/woocommerce-blocks/assets/js/blocks/product-filters/edit.tsx

```diff
@@ -2,78 +2,63 @@
-import { Template } from '@wordpress/blocks';
+import { BlockEditProps } from '@wordpress/blocks';
 import { useSelect } from '@wordpress/data';
-import { getSetting } from '@woocommerce/settings';
-import type { AttributeSetting } from '@woocommerce/types';
 
 /**
  * Internal dependencies
  */
-import { EditProps, FilterType } from './types';
-import { getAllowedBlocks } from './utils';
 import Downgrade from './components/downgrade';
 import Warning from './components/warning';
 import './editor.scss';
 
-const DISALLOWED_BLOCKS = [
-	'woocommerce/filter-wrapper',
-	'woocommerce/product-filters',
-];
-
-const ATTRIBUTES = getSetting< AttributeSetting[] >( 'attributes', [] );
-
-const firstAttribute = ATTRIBUTES.find( Boolean );
-
-const templates: Partial< Record< FilterType, Template[] > > = {
-	'active-filters': [ [ 'woocommerce/product-filters-active', {} ] ],
-	'price-filter': [ [ 'woocommerce/product-filters-price', {} ] ],
-	'stock-filter': [ [ 'woocommerce/product-filters-stock-status', {} ] ],
-	'rating-filter': [ [ 'woocommerce/product-filters-rating', {} ] ],
-	'product-filters': [
-		[ 'woocommerce/product-filters-active', {} ],
-		[ 'woocommerce/product-filters-price', {} ],
-		[ 'woocommerce/product-filters-stock-status', {} ],
-		[ 'woocommerce/product-filters-rating', {} ],
-	],
+const BLOCK_NAME_MAP = {
+	'active-filters': 'woocommerce/product-filters-active',
+	'price-filter': 'woocommerce/product-filters-price',
+	'stock-filter': 'woocommerce/product-filters-stock-status',
+	'rating-filter': 'woocommerce/product-filters-rating',
+	'attribute-filter': 'woocommerce/product-filters-attribute',
 };
 
-if ( firstAttribute ) {
-	templates[ 'attribute-filter' ] = [
-		[
-			'woocommerce/product-filters-attribute',
-			{ attributeId: parseInt( firstAttribute?.attribute_id, 10 ) },
-		],
-	];
-
-	templates[ 'product-filters' ]?.push( [
-		'woocommerce/product-filters-attribute',
-		{ attributeId: parseInt( firstAttribute?.attribute_id, 10 ) },
-	] );
-}
-
-const Edit = ( props: EditProps ) => {
-	const allowedBlocks = getAllowedBlocks( DISALLOWED_BLOCKS );
-
-	const template = templates[ props.attributes.filterType ];
+type FilterType = keyof typeof BLOCK_NAME_MAP;
 
+const Edit = ( {
+	attributes,
+	clientId,
+}: BlockEditProps< { heading: string; filterType: FilterType } > ) => {
 	const blockProps = useBlockProps();
 
 	const isNested = useSelect( ( select ) => {
 		const { getBlockParentsByBlockName } = select( 'core/block-editor' );
 		return !! getBlockParentsByBlockName(
-			props.clientId,
+			clientId,
 			'woocommerce/product-collection'
 		).length;
 	} );
 
 	return (
 		<nav { ...blockProps }>
 			{ ! isNested && <Warning /> }
-			<Downgrade clientId={ props.clientId } />
+			<Downgrade
+				filterType={ attributes.filterType }
+				clientId={ clientId }
+			/>
 			<InnerBlocks
-				template={ template }
-				allowedBlocks={ allowedBlocks }
+				allowedBlocks={ [ 'core/heading' ] }
```

Comment:
The wrapper block should accept other blocks than only the `core/heading` for more complex layout support. All filter blocks are still needed to be excluded.

Reply:
So I did this, I just want to note one behaviour that I'm guessing is behaviour of allowed blocks API, when you nest inside `product-collection` even though, `product-collection` is not allowed child by my function (I double checked this) you can still insert product-collection as child.